cachi2/core/package_managers/bundler/parser.py

```diff
@@ -226,7 +226,24 @@ def parse_lockfile(package_dir: RootedPath) -> ParseResult:
     result: ParseResult = []
     for dep in dependencies:
         if dep["type"] == "rubygems":
-            result.append(GemDependency(**dep))
+            if dep["platform"] == "ruby":
+                result.append(GemDependency(**dep))
+            else:
+                full_name = "-".join([dep["name"], dep["version"], dep["platform"]])
+                log.info("Found a binary dependency %s", full_name)
+                if allow_binary:
+                    log.warning(
+                        "Will download binary dependency %s because 'allow_binary' is set to True",
+                        full_name,
+                    )
+                    result.append(GemPlatformSpecificDependency(**dep))
+                else:
+                    # No need to force a platform if we skip the packages.
+                    log.warning(
+                        "Skipping binary dependency %s because 'allow_binary' is set to False."
+                        " This will likely result in an unbuildable package.",
```

Comment:
```suggestion
log.warning("Skipping platform-specific gem %s because 'allow_binary' option is disabled", full_name)
```

Reply:
I would rather give users another clue. There are many ways a build could go wrong, messing with binary dependencies is just one of those. I would expect a user to start debugging a failing build with going through logs, thus putting clues right there seems reasonable. Besides that our exceptions often have "solution" field, I think having a detailed warning is in the spirit of having solutions for exceptions.

Reply:
This warning debate has IMO gone out of hand a bit, let's just agree on keeping it, it's just a string of characters after all that doesn't decrease the quality of the output and can potentially only improve it.